be/test/exec/jni_scanner_test.cpp

```diff
@@ -0,0 +1,317 @@
+        c1.type.field_names.emplace_back("Cc1");
+
+        SlotDesc slot_descs[] = {c0, c1, {""}};
+        return create_tuple_desc(slot_descs);
+    }
+
+    void create_runtime_state(const std::string& timezone) {
+        TUniqueId fragment_id;
+        TQueryOptions query_options;
+        TQueryGlobals query_globals;
+        if (timezone != "") {
+            query_globals.__set_time_zone(timezone);
+        }
+        _runtime_state = _pool.add(new RuntimeState(fragment_id, query_options, query_globals, nullptr));
+        _runtime_state->init_instance_mem_tracker();
+    }
+
+    ObjectPool _pool;
+    RuntimeState* _runtime_state;
+};
+
+TEST_F(JniScannerTest, test_create_paimon_jni_scanner) {
+    // create jni scanner
+    JniScanner::CreateOptions options;
+
+    TTableDescriptor t_table_desc;
+    thrift_read_from_file(&t_table_desc, "./be/test/exec/test_data/jni_scanner/test_create_scanner/paimon.table.json");
+    PaimonTableDescriptor paimonTable(t_table_desc, &_pool);
+    options.hive_table = &paimonTable;
+
+    THdfsScanRange t_hdfs_scan_range;
+    thrift_read_from_file(&t_hdfs_scan_range,
+                          "./be/test/exec/test_data/jni_scanner/test_create_scanner/paimon.scan.json");
+    options.scan_range = &t_hdfs_scan_range;
+
+    FSOptions fs_options;
+    init_fs_options(&fs_options);
+    options.fs_options = &fs_options;
+
+    JniScanner* scanner = create_paimon_jni_scanner(options);
+
+    // update columns.
+    TupleDescriptor* tuple_desc = create_default_tuple_desc();
+    init_hdfs_scanner_context(&(scanner->_scanner_ctx), tuple_desc);
+    scanner->update_jni_scanner_params();
+
+    // check parameters.
+    for (const auto& kv : scanner->_jni_scanner_params) {
+        if (kv.second.size() < 128) {
+            std::cout << "{\"" << kv.first << "\", \"" << kv.second << "\"},\n";
+        }
+    }
+}
+
+TEST_F(JniScannerTest, test_create_hudi_jni_scanner) {
+    // create jni scanner
+    JniScanner::CreateOptions options;
+
+    TTableDescriptor t_table_desc;
+    thrift_read_from_file(&t_table_desc, "./be/test/exec/test_data/jni_scanner/test_create_scanner/hudi.table.json");
+    HudiTableDescriptor hudiTable(t_table_desc, &_pool);
+    options.hive_table = &hudiTable;
+
+    THdfsScanRange t_hdfs_scan_range;
+    thrift_read_from_file(&t_hdfs_scan_range,
+                          "./be/test/exec/test_data/jni_scanner/test_create_scanner/hudi.scan.json");
+    options.scan_range = &t_hdfs_scan_range;
+
+    FSOptions fs_options;
+    init_fs_options(&fs_options);
+    options.fs_options = &fs_options;
+
+    JniScanner* scanner = create_hudi_jni_scanner(options);
+
+    // update columns.
+    TupleDescriptor* tuple_desc = create_default_tuple_desc();
+    init_hdfs_scanner_context(&(scanner->_scanner_ctx), tuple_desc);
+    scanner->update_jni_scanner_params();
+
+    // check parameters.
+    for (const auto& kv : scanner->_jni_scanner_params) {
```

Comment:
this is not any machine check, but human manual check. need consider how to check the correctness in UT, otherwise the UT looks useless.

Reply:
Dear, the main purpose of this ut is just for coverage. But you are right, I should add some checks.

Reply:
I've added some checks